macos/FluentUI/Button/Button.swift

```diff
@@ -335,6 +316,25 @@ open class Button: NSButton {
 		path.fill()
 	}
 
+	open override func viewDidChangeEffectiveAppearance() {
+		super.viewDidChangeEffectiveAppearance()
+
+		// At this point in code, the Appearance of the View has been changed either to darkAqua or
+		// aqua and since this button has Shadow Layers, it relies on updateLayer() getting called
+		// immediately after this function so that all the layers get the correct backgroundColors.
+		// However, based on how layer-backed views operate, updateLayer() isn't called immediately
+		// and is instead backloaded, to be called asynchronously along with all other UI updates
+		// already queued up, at a time determined by the system APIs. Call displayIfNeeded here to
+		// immediately invoke an updateLayer() call. This addresses a specific case where the
+		// Button's container view performs a
+		// (1) setAppearance, (which then triggers a viewDidChangeEffectiveAppearance call)
+		// immediately followed by a
+		// (2) window.toggleFullScreen
+		// where we need the appearance change effects on the layers to to take place BEFORE the
+		// FullScreen toggle, to prevent an overlap in animation changes in Button background colors.
+		displayIfNeeded()
+	}
```

Comment:
Kinda feels like the issue is in the usage which needs the synchronous display pass and we shouldn't enforce this on every user of the button? Could the user of this button call displayIfNeeded on the button?